cpp/connectivity_module/connectivity_module.cpp

```diff
@@ -1,136 +1,95 @@
 #include <queue>
 #include <unordered_map>
 
-#include "mg_procedure.h"
-#include "utils/on_scope_exit.hpp"
+#include <mg_exceptions.hpp>
+#include <mg_procedure.h>
+#include <mg_utils.hpp>
+
+namespace {
+
+const char *field_component_id = "component_id";
+const char *field_vertex = "node";
+
+void InsertWeaklyComponentResult(const mgp_graph *graph, mgp_result *result,
+                                 mgp_memory *memory, const int component_id,
+                                 const int vertex_id) {
+  mgp_result_record *record = mgp_result_new_record(result);
+  if (record == nullptr) {
+    throw mg_exception::NotEnoughMemoryException();
+  }
+
+  mg_utility::InsertNodeValueResult(graph, record, field_vertex, vertex_id,
+                                    memory);
+
+  mg_utility::InsertIntValueResult(record, field_component_id, component_id,
+                                   memory);
+}
 
 /// Finds weakly connected components of a graph.
 ///
 /// Time complexity: O(|V|+|E|)
-static void weak(const mgp_list *args, const mgp_graph *graph,
+static void Weak(const mgp_list *args, const mgp_graph *memgraph_graph,
                  mgp_result *result, mgp_memory *memory) {
-  std::unordered_map<int64_t, int64_t> vertex_component;
-  mgp_vertices_iterator *vertices_iterator =
-      mgp_graph_iter_vertices(graph, memory);
-  if (vertices_iterator == nullptr) {
-    mgp_result_set_error_msg(result, "Unable to create vertices iterator");
-    return;
-  }
-  utils::OnScopeExit vertex_iterator_deleter([&vertices_iterator] {
-    if (vertices_iterator != nullptr) {
-      mgp_vertices_iterator_destroy(vertices_iterator);
-    }
-  });
-
-  int64_t curr_component = 0;
-  for (const mgp_vertex *vertex = mgp_vertices_iterator_get(vertices_iterator);
-       vertex != nullptr;
-       vertex = mgp_vertices_iterator_next(vertices_iterator)) {
-    mgp_vertex_id vertex_id = mgp_vertex_get_id(vertex);
-    if (vertex_component.find(vertex_id.as_int) != vertex_component.end())
-      continue;
-
-    // Run BFS from current vertex.
-    std::queue<int64_t> q;
-    q.push(vertex_id.as_int);
-    vertex_component[vertex_id.as_int] = curr_component;
-    while (!q.empty()) {
-      mgp_vertex *v = mgp_graph_get_vertex_by_id(graph, {q.front()}, memory);
-      if (v == nullptr) {
-        mgp_result_set_error_msg(result, "Unable to get vertex");
-        return;
-      }
-      utils::OnScopeExit v_deleter([&v] {
-        if (v != nullptr) {
-          mgp_vertex_destroy(v);
-        }
-      });
+  try {
+    mg_graph::Graph<> *graph =
+        mg_utility::GetGraphView(memgraph_graph, result, memory);
 
-      q.pop();
+    std::unordered_map<uint64_t, uint64_t> vertex_component;
+    int64_t curr_component = 0;
+    for (mg_graph::Node vertex : graph->Nodes()) {
```

Comment:
here also, use `auto`
```suggestion
    for (auto vertex : graph->Nodes()) {
```